src/typechecker.rs

```diff
@@ -2805,20 +2899,21 @@ pub fn typecheck_expression(
                                                                         .find(|f| f.name == name)
                                                                         .map(|f| f.ty);
 
-                                                                    let field_type =
-                                                                        if let Some(field_type) =
-                                                                            field_type
-                                                                        {
-                                                                            Some(
-                                                                        substitute_typevars_in_type(
-                                                                            field_type,
-                                                                            &generic_parameters,
-                                                                            project,
-                                                                        ),
-                                                                    )
-                                                                        } else {
-                                                                            field_type
-                                                                        };
+                                                                    let field_type = if let Some(
+                                                                        field_type,
+                                                                    ) =
+                                                                        field_type
+                                                                    {
+                                                                        Some(
+                                                                                substitute_typevars_in_type(
+                                                                                    field_type,
+                                                                                    &generic_parameters,
+                                                                                    project,
+                                                                                ),
+                                                                            )
+                                                                    } else {
+                                                                        field_type
+                                                                    };
```

Comment:
These appear to be identical, if that's really the case please consider moving this and the whitespace changes above into their own commit - not sure what's happening here, this both looks incredibly broken to me, but I'm sure rustfmt knows what it's doing :shrug:

Reply:
I'll see if I can make rustfmt not create this weirdo thing